gtsam/nonlinear/NonlinearConjugateGradientOptimizer.h

```diff
@@ -40,10 +38,10 @@ class GTSAM_EXPORT NonlinearConjugateGradientOptimizer : public NonlinearOptimiz
     System(const NonlinearFactorGraph &graph) :
         graph_(graph) {
     }
-    double error(const State &state) const;
-    Gradient gradient(const State &state) const;
-    State advance(const State &current, const double alpha,
-        const Gradient &g) const;
+    double error(const Values &state) const;
```

Comment:
I disagree with this change. Other systems have different “State” and I would like to keep the naming uniform.

Reply:
Okay, but I propose moving the `System` class outside of `NonlinearConjugateGradientOptimizer` so they are decoupled. I am not sure how you would update `System` in this case.